monkey/common/agent_configuration/agent_sub_configurations.py

```diff
@@ -118,6 +118,15 @@ class ExploitationConfiguration:
 
 @dataclass(frozen=True)
 class PropagationConfiguration:
+    """
+    A configuration for propagation
+
+    Attributes:
+        :param maximum_depth: Maximum number of hops to take from the Island server
```

Comment:
I think we can do a better job describing this. Take a look in the documentation and see what we have there.

Reply:
How about now?